### PR TITLE
Patient Profile Page

### PR DESCRIPTION
**Pull request by hassanelnajjar**
# Patient Profile Page
- [x] add patient-history table component and create patient-profile-page and pass the history data to patient history component
- [x] Add patient-detail form component
- [x] Add patient-PatientTreatmentForm component
- [x] Add Loading common component
- [x] modify the min-hight at root div and add overflow = hidden to it
- [x] modify the with of children inside sidebar component to be 100%

relates #46

### CODE REVIEW

File: client/src/components/common/Loading/style.css [hunk 1]
```diff
@@ -0,0 +1,6 @@
+.spin-style{
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  width: 100%;
+}
```
Comment (Mu7ammadAbed) [inline client/src/components/common/Loading/style.css:6]: new line here

File: client/src/components/common/Title/style.css [hunk 2]
```diff
@@ -0,0 +1,4 @@
+.custom-title{
+  text-align: center;
+  padding: 2rem 0;
+}
```
Comment (Mu7ammadAbed) [inline client/src/components/common/Title/style.css:4]: new line here

File: client/src/components/common/Title/index.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import { Typography } from 'antd';
+
+import './style.css';
+import { string } from 'prop-types';
+
+const { Title } = Typography;
+
+const CustomTitle = ({ text }) => (
+  <Title className="custom-title" level={1}>
```
Comment (Mu7ammadAbed) [inline client/src/components/common/Title/index.js:10]:
```suggestion
  <Title className="custom-title" level={2}>
```

File: client/src/components/common/Loading/style.css (hunk 1 above)
Comment (hassanelnajjar) [reply client/src/components/common/Loading/style.css:6]: Done

File: client/src/components/common/Title/style.css (hunk 2 above)
Comment (hassanelnajjar) [reply client/src/components/common/Title/style.css:4]: Done

File: client/src/components/common/Title/index.js (hunk 3 above)
Comment (hassanelnajjar) [reply client/src/components/common/Title/index.js:10]: Done

File: client/src/components/PatientHistory/index.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+const columns = ['Description', 'Price', 'Payment', 'Remaining', 'Time'].map(
+  (column) => ({
+    title: column,
+    dataIndex: column,
+    key: column,
+  })
+);
+
+function PatientHistory({ historyData }) {
```
Comment (Mu7ammadAbed) [inline client/src/components/PatientHistory/index.js:14]: Arrow function :)

File: client/src/pages/admin/PatientProfile/index.js [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+        setLoading(false);
+        if (err.response) {
+          const {
+            response: { data },
+          } = err;
+          return setErrorMessage(data.message ? data.message : data);
+        }
+        return setErrorMessage(err);
+      }
+    };
```
Comment (Mu7ammadAbed) [inline client/src/pages/admin/PatientProfile/index.js:43]: Move this function to outside the useEffect

Comment (hassanelnajjar) [reply client/src/pages/admin/PatientProfile/index.js:43]: Done

File: client/src/components/PatientHistory/index.js (hunk 4 above)
Comment (hassanelnajjar) [reply client/src/components/PatientHistory/index.js:14]: Done

**Review by Mu7ammadAbed — APPROVED** [review]

**Review by Mu7ammadAbed — APPROVED** [review]

**Review by AlaaShurrab — APPROVED** [review]

**Pull request merged**